### ISSUE TITLE
gitlib-3.1.0.1 does not compile

### ISSUE THREAD

**Issue by peti** [jwiegley/gitlib#46]
Citing from http://hydra.cryp.to/build/835780/log/raw:

```
Git/Blob.hs:53:69:
    Couldn't match type ‘Filesystem.Path.CurrentOS.FilePath’
                   with ‘[Char]’
    Expected type: Prelude.FilePath
      Actual type: Filesystem.Path.CurrentOS.FilePath
    In the first argument of ‘sinkFile’, namely ‘(decodeString path)’
    In the second argument of ‘($$)’, namely
      ‘sinkFile (decodeString path)’

Git/Blob.hs:54:59:
    Couldn't match type ‘Filesystem.Path.CurrentOS.FilePath’
                   with ‘[Char]’
    Expected type: Prelude.FilePath
      Actual type: Filesystem.Path.CurrentOS.FilePath
    In the first argument of ‘sinkFile’, namely ‘(decodeString path)’
    In the second argument of ‘($$)’, namely
      ‘sinkFile (decodeString path)’

Git/Blob.hs:55:59:
    Couldn't match type ‘Filesystem.Path.CurrentOS.FilePath’
                   with ‘[Char]’
    Expected type: Prelude.FilePath
      Actual type: Filesystem.Path.CurrentOS.FilePath
    In the first argument of ‘sinkFile’, namely ‘(decodeString path)’
    In the second argument of ‘($$)’, namely
      ‘sinkFile (decodeString path)’
```

**Comment by nomeata** [comment]
I also started seeing this. Weirdly, it compiled before, with the same versions of gitlib and system-filepath: https://s3.amazonaws.com/archive.travis-ci.org/jobs/61955244/log.txt

Any idea what package caused this to break?

**Comment by nomeata** [comment]
Ah, it seem that conduit-combinators was the culprit: http://hdiff.luite.com/cgit/conduit-combinators/commit/?id=864fbe129d2dfc8d026bc109223057c9f7ba9b3a

**Comment by snoyberg** [comment]
Explanation of the change is here, sorry for the breakage: https://github.com/yesodweb/yesodweb.com-content/blob/master/blog/2015/05/deprecating-system-filepath.md

@nomeata Hopefully no users are affected by sudden breakage since they're all using a curated system like Nix, Debian, or LTS Haskell ;)

**Comment by nomeata** [comment]
And who cures about the curators :-]

**Comment by snoyberg** [comment]
You means cures about carators, right? ;)

**Comment by nomeata** [comment]
Who cores what I mean

**Comment by mdorman** [comment]
Shoule be resolved as of 295b36bb098a7da771e21c7b263c346c24d0d7d3
